driver/src/contracts/gnosis_safe_gas_station.rs

```diff
@@ -0,0 +1,82 @@
+use ethcontract::U256;
+use serde::Deserialize;
+use std::time::Duration;
+use uint::FromDecStrErr;
+
+/// Result of https://safe-relay.gnosis.io/ api call `/v1/gas-station`.
+///
+/// Prices are in wei.
+#[derive(Deserialize, Debug, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GasPrice {
+    pub last_update: String,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub lowest: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub safe_low: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub standard: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fast: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fastest: U256,
+}
+
+/// Retrieves gas prices from the Gnosis Safe Relay api.
+///
+/// Uses a timeout of 10 seconds.
+pub fn get_gas_price() -> Result<GasPrice, reqwest::Error> {
+    const URL: &str = "https://safe-relay.gnosis.io/api/v1/gas-station/";
+    const TIMEOUT: Duration = Duration::from_secs(10);
+    // It would be more efficient to reuse the client between calls. However, we
+    // only call this function once per batch when submitting a solution so this
+    // is not important at the moment.
+    let client = reqwest::blocking::ClientBuilder::new()
+        .timeout(TIMEOUT)
+        .build()?;
+    client.get(URL).send()?.json()
+}
+
+fn deserialize_u256_from_string<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    U256::from_dec_str(&s)
+        .map_err(uint_error_to_string)
```

Comment:
nit: why not use debug format from `uint::FromDecStrErr` directly? Should we maybe also include the bad string in the error to make debugging easier?

Reply:
Included the source string in the message in new commit.
In practice `Debug` would work but conceptually it isn't meant for the error message so it felt better to do it like this.